compiler/x/codegen/OMRRegisterDependency.cpp

```diff
@@ -1151,55 +1042,8 @@ void TR_X86RegisterDependencyGroup::assignFPRegisters(TR::Instruction   *prevIns
 
    if (numberOfRegisters > 0)
       {
-      TR::Register *reqdTopVirtReg = NULL;
-
       int32_t i;
```

Comment:
This can be inlined as well.

Reply:
I don't believe so.  It is used in multiple loops.

Reply:
Right, but each loop initializer sets it to 0, so we are needlessly extending the scope of this variable to the entire function, rather than each individual loop. Is there a location we rely on the value of `i` outside of a loop that is not zero initializes (`for (i = 0, ...`)?

Reply:
Sorry--I get the reasoning, but somehow I believed that you couldn't declare the iteration variable multiple times in different loops.  I tried a simple test to confirm that you can, so I can make that change.

Reply:
Your inner C99 wizardry is coming to life. In C99 it is not possible, but in C++ it is. Thanks for making the change.